plugins/inputs/x509_cert/java_key_store_test.go

```diff
@@ -0,0 +1,229 @@
+		path = strings.TrimPrefix(path, "pkcs12://")
+		path = strings.TrimPrefix(path, "jks://")
+		require.NoError(t, os.Chmod(path, mode))
+	}
+}
+
+// createTestPKCS12 creates a temporary PKCS#12 keystore
+func createTestPKCS12(t *testing.T, certPEM, keyPEM []byte) string {
+	t.Helper()
+
+	// Decode certificate
+	block, _ := pem.Decode(certPEM)
+	require.NotNil(t, block, "failed to parse certificate PEM")
+
+	cert, err := x509.ParseCertificate(block.Bytes)
+	require.NoError(t, err)
+
+	// Decode private key
+	block, _ = pem.Decode(keyPEM)
+	require.NotNil(t, block, "failed to parse private key PEM")
+
+	privKey, err := x509.ParsePKCS1PrivateKey(block.Bytes)
+	require.NoError(t, err)
+
+	// Encode PKCS#12 keystore
+	pfxData, err := pkcs12.Modern.Encode(privKey, cert, nil, "test-password")
+	require.NoError(t, err)
+
+	// Use `t.TempDir()` to ensure cleanup
+	tempDir := t.TempDir()
+	pkcs12Path := filepath.Join(tempDir, "test-keystore.p12")
+
+	err = os.WriteFile(pkcs12Path, pfxData, 0600)
+	require.NoError(t, err)
+
+	pkcs12Path = filepath.ToSlash(pkcs12Path)
+	if runtime.GOOS == "windows" {
+		pkcs12Path = strings.TrimPrefix(pkcs12Path, "/")
+	}
+
+	return "pkcs12://" + pkcs12Path
+}
+
+// createTestJKS creates a temporary JKS keystore
+func createTestJKS(t *testing.T, certDER []byte) string {
+	t.Helper()
+
+	// Use `t.TempDir()` to ensure cleanup
+	tempDir := t.TempDir()
+	jksPath := filepath.Join(tempDir, "test-keystore.jks")
+
+	// Create JKS keystore and add a trusted certificate
+	jks := keystore.New()
+	err := jks.SetTrustedCertificateEntry("test-alias", keystore.TrustedCertificateEntry{
+		Certificate: keystore.Certificate{
+			Type:    "X.509",
+			Content: certDER,
+		},
+	})
+	require.NoError(t, err)
+
+	// Write keystore to file
+	output, err := os.Create(jksPath)
+	require.NoError(t, err)
+	defer output.Close()
+
+	require.NoError(t, jks.Store(output, []byte("test-password")))
+
+	jksPath = filepath.ToSlash(jksPath)
+	if runtime.GOOS == "windows" {
+		jksPath = strings.TrimPrefix(jksPath, "/")
+	}
+
+	return "jks://" + jksPath
+}
+
+func TestGatherKeystores_Success(t *testing.T) {
+	pkcs12Path, jksPath := generateTestKeystores(t)
+
+	tests := []struct {
+		name     string
+		mode     os.FileMode
+		content  string
+		password string
+	}{
+		{name: "valid PKCS12 keystore", mode: 0640, content: pkcs12Path, password: "test-password"},
+		{name: "valid JKS keystore", mode: 0640, content: jksPath, password: "test-password"},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			setFilePermissions(t, test.content, test.mode)
+
+			sc := X509Cert{
+				Sources: []string{test.content},
+				Log:     testutil.Logger{},
+			}
+
+			if test.password != "" {
+				sc.Password = config.NewSecret([]byte(test.password))
+			} else {
+				sc.Password = config.NewSecret(nil)
+			}
+
+			require.NoError(t, sc.Init())
+
+			acc := testutil.Accumulator{}
+			err := sc.Gather(&acc)
+
+			require.NoError(t, err)
```

Comment:
The password is never empty in those tests and we can save some more lines with
```suggestion
			plugin := X509Cert{
				Sources:  []string{test.content},
				Password: config.NewSecret([]byte(test.password)),
				Log:      testutil.Logger{},
			}
			require.NoError(t, plugin.Init())

			var acc testutil.Accumulator
			require.NoError(t, plugin.Gather(&acc))
```